File: mortgage-calculator/src/Calculator.js
import React, { useState } from "react";

export default function Calculator(props) {
    const [homePriceState, setHomePriceState] = useState(0);
    const [downPaymentState, setDownPaymentState] = useState(0);
    const [interestRateState, setInterestRateState] = useState(0);
    const [termState, setTermState] = useState(0);


    const inputs = [{title:"Home Price", id:"homePrice", value:homePriceState, setState:setHomePriceState},
        {title:"Down Payment", id:"downPayment", value:downPaymentState, setState:setDownPaymentState},
        {title:"Interest Rate", id:"interestRate", value:interestRateState, setState:setInterestRateState},
        {title:"Term", id:"term", value:termState, setState:setTermState}]

    function handleChange(event, setStateFunction){
        setStateFunction(event.target.value)
    }

    const monthlyInterestRate = interestRateState/12
    const termInMonths = termState *12
    const propertyTaxes = (.01608*homePriceState)/12
    const insurance = 184
    const inputsMap = inputs.map((input)=><div><label for={input.id}>{input.title}</label><input id={input.id} onChange={(e)=>handleChange(e,input.setState)}/></div>)

    const principal = homePriceState - downPaymentState

    const mortgageBeforeTaxesAndInsurance = (((monthlyInterestRate*(1+monthlyInterestRate)**termInMonths))/(((1+monthlyInterestRate)**termInMonths)-1))*principal

    const mortgageAfterTaxesAndInsurance = mortgageBeforeTaxesAndInsurance + insurance + propertyTaxes

    return(
        <div>
            {inputsMap}
            {mortgageAfterTaxesAndInsurance}
        </div>
        
    )
}

File: calculator.py
homePrice = int(input("Enter the price of the home: "))
interestRate = float(input("Enter the interest rate: "))/12
downPayment = int(input("Enter your down payment: "))
# homePrice = 400000
# interestRate = .05875/12
# downPayment = 90000

insurance = 184
propertyTaxes = (.01608*homePrice)/12
monthlyIncome = 9000
term = 360

principal = homePrice - downPayment

mortgageBeforeTaxesAndInsurance = (((interestRate*(1+interestRate)**term))/(((1+interestRate)**term)-1))*principal

mortgageAfterTaxesAndInsurance = mortgageBeforeTaxesAndInsurance + insurance + propertyTaxes

percentOfMonthlyIncome = mortgageAfterTaxesAndInsurance/monthlyIncome
maxPayment = monthlyIncome*.28



print("\n"+"Results"+"---------------------"+"\n"
      "Payment: $"+str(mortgageAfterTaxesAndInsurance) + "\n" + 
      "Percent of Monthly Payment: " + str(percentOfMonthlyIncome*100)+"%" + "\n" +
      "Affordable: "+str(percentOfMonthlyIncome<=.28) + "\n" +
      "Max Affordable Payment Per Month: $" +str(maxPayment) + "\n")